pkgs/development/haskell-modules/hackage-packages.nix

```diff
@@ -222232,7 +222251,7 @@ self: {
      }) {};
 
   "polysemy-zoo" = callPackage
-    ({ mkDerivation, async, base, compact, constraints, containers
+    ({ mkDerivation, async, base, compact, constraints_0_12, containers
```

Comment:
You'll have to explicitly override `constraints` in `configuration-common.nix`.

Here's an example:

https://github.com/NixOS/nixpkgs/blob/599582499c548c8ad7f95dc0b2cb4d276795abeb/pkgs/development/haskell-modules/configuration-common.nix#L1478